controllers/workspace/devworkspace_controller.go

```diff
@@ -177,9 +177,17 @@ func (r *DevWorkspaceReconciler) Reconcile(req ctrl.Request) (reconcileResult ct
 		return reconcile.Result{}, nil
 	}
 	workspace.Spec.Template = *flattenedWorkspace
+
+	storageProvisioner, err := storage.GetProvisioner(workspace)
```

Comment:
I see the issue we may have with an ability to change storage type.
As far as I understand, user may have devworkspace with any valid storage, devworkspace will get finalizers, but then after user sets invalid storage type, they'll get workspace which will not be possible to delete, until storage type is not fixed.

If you think it will work in that way, please create an issue or let me know, I'll do.

P.S. It's not related to these changes in any case, just observation when we may take care of to improve stability.

Reply:
Yeah, we need to improve webhooks to forbid making changes to storage type. Hosted Che had this problem, IIRC, where you could set a persistent workspace to ephemeral and the storage wouldn't necessarily be cleaned up. 

I think the safe bet is to block all changes to storage -- what if async storage eventually decides to use a different PVC than common? If we eventually implement per-workspace storage, we'll also run into this problem.